superdesk/storage/amazon/amazon_media_storage.py

```diff
@@ -59,11 +60,20 @@ def url_for_media(self, media_id):
         endpoint = 's3-%s.amazonaws.com' % self.region
         return '%s://%s.%s/%s' % (protocol, self.container_name, endpoint, media_id)
 
+    def name_for_media(self, media_id):
+        if not self.app.config.get('AMAZON_SERVE_DIRECT_LINKS', False):
+            return media_id
+
+        return '%s/%s' % (time.strftime('%Y%m%d'), media_id)
+
     def read_from_config(self):
         self.region = self.app.config.get('AMAZON_REGION', 'us-east-1') or 'us-east-1'
         username = self.app.config['AMAZON_ACCESS_KEY_ID']
         api_key = self.app.config['AMAZON_SECRET_ACCESS_KEY']
         self.container_name = self.app.config['AMAZON_CONTAINER_NAME']
+        self.kwargs = {}
+        if self.app.config.get('AMAZON_SERVE_DIRECT_LINKS', False):
```

Comment:
maybe this should be `PUBLIC_LINKS` not `DIRECT_LINKS` it would be more obvious imo